pkgs/development/libraries/xgboost/default.nix

```diff
@@ -27,11 +28,10 @@ let
   # This ensures xgboost gets the correct libstdc++ when
   # built with cuda support. This may be removed once
   # #226165 rewrites cudaStdenv
-  inherit (cudaPackages) backendStdenv;
-  stdenv = if cudaSupport then backendStdenv else inputs.stdenv;
+  effectiveStdenv = if cudaSupport then cudaPackages.backendStdenv else stdenv;
```

Comment:
This is effectively a no-op, just uses a new name instead of hiding the old symbol. I think this generates a bigger diff, but I've no preferences. If you'd like to go with the new name, I'd still hide the old symbol with `stdenv = throw "Use effectiveStdenv <...>"`

Reply:
Addressed in latest